executor/insert_common.go

```diff
@@ -488,12 +488,7 @@ func (e *InsertValues) adjustAutoIncrementDatum(d types.Datum, hasValue bool, c
 		d.SetNull()
 	}
 	if !d.IsNull() {
-		sc := e.ctx.GetSessionVars().StmtCtx
```

Comment:
you should not remove these codes, need to convert it to the field type of the column.

Reply:
The converting is duplicate which has been done in: https://github.com/pingcap/tidb/blob/master/executor/insert_common.go#L252